### ISSUE TITLE
Select with multiple attribute not working

### ISSUE THREAD

**Issue by cebbinger** [jeremyHixon/RationalOptionPages#19]
I tried to create a select field that allows the user to choose multiple entries.
The Input will not be saved and no option will be selected by default.

```php
//...
'fields' => array(
  'test_field'    => array(
    'id'         => 'test_field',
    'title'      => 'Test Field',
    'type'       => 'select',
    'choices'    => array( 'a', 'b', 'c' ),
    'attributes' => array(
      'multiple'   => true,
    ),
  ),
//...
```

Everything works fine without `'multiple' => true`.
Am I missing something here or is it a bug?

**Comment by stfn00** [comment]
Also for me.

**Comment by jeremyHixon** [comment]
Investigating

**Comment by jeremyHixon** [comment]
Just added an update. Should fix the "select" issue.
cd8106a9cbf9191b197c0d0acafe9a687675c698

**Comment by stfn00** [comment]
Hi Jeremy,
i did some tests but it still doesn't work, i.e. it keeps saving the field as an empty string.

After a little look at the class, I think I've solved the problem.
In the `sanitize_setting()` function, I made sure to sanitize the `select` with `multiple` attribute giving it the base value as an array and not as in the `case` of `default` where it is sanitized in string.

After this change, both `multiple` and non `select` works.

I proceeded to send you a pull request. I wait your feedback.

Best regards,
Stefano.
